tests/ShadowDOM/js/Range.js

```diff
@@ -11,76 +11,525 @@
 suite('Range', function() {
 
   var wrap = ShadowDOMPolyfill.wrap;
+  var unwrap = ShadowDOMPolyfill.unwrap;
+  var wrapIfNeeded = ShadowDOMPolyfill.wrapIfNeeded;
+  var isNativeShadowDomSupported;
 
-  var div;
+  var hosts;
+  var customElementPrefix = "range-custom-element-";
+  var nativeCustomElementPrefix = "range-native-element-";
+  var customElementIndex=0;
+  var nativeCustomElementIndex=0;
 
-  teardown(function() {
-    if (div && div.parentNode)
-      div.parentNode.removeChild(div);
-    div = undefined;
+  setup(function() {
+    isNativeShadowDomSupported = !!unwrap(document.createElement('div')).createShadowRoot;
```

Comment:
ah, clever. So you can compare native to polyfilled. That's neat.

one thing I wanted to do at some point, was make all of the SD polyfill tests work on native too. That way we can compare. I wouldn't necessarily assume native is right, if they differ, but would be nice to have a test suite that works on both.

anyway no action needed, just wanted to share some thougths